Library/OcCryptoLib/Streebog.c

```diff
@@ -0,0 +1,378 @@
+/** @file
+  Copyright (c) 2013, Alexey Degtyarev <alexey@renatasystems.org>. All rights reserved.<BR>
+  Copyright (c) 2022 Maxim Kuznetsov. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause
+**/
+
+#ifdef OC_CRYPTO_SUPPORTS_STREEBOG
+
+#include "CryptoInternal.h"
+#include "Streebog.h"
+
+#define BSWAP64(x) \
+    (((x & 0xFF00000000000000ULL) >> 56) | \
+     ((x & 0x00FF000000000000ULL) >> 40) | \
+     ((x & 0x0000FF0000000000ULL) >> 24) | \
+     ((x & 0x000000FF00000000ULL) >>  8) | \
+     ((x & 0x00000000FF000000ULL) <<  8) | \
+     ((x & 0x0000000000FF0000ULL) << 24) | \
+     ((x & 0x000000000000FF00ULL) << 40) | \
+     ((x & 0x00000000000000FFULL) << 56))
+
+VOID
+GOST34112012Cleanup (
+  STREEBOG_CONTEXT  *Context
+  )
+{
+  for (INT32 i = 0; i < 64; ++i) {
```

Comment:
EDK-II convention follows the rule where variables are declared at the beginning of functions.